scenes/vehicle/car_audio_controller.gd

```diff
@@ -0,0 +1,179 @@
+	car_audio_player_engine_off.volume_db = -40
+	car_audio_player_engine_off.play()
+
+	car_audio_player_plop.stream = stream_plop
+	
+	car_audio_player_wind.stream = stream_drag_wind
+	
+	car_audio_player_brakes.stream = stream_brakes
+	
+	car_audio_player_gear_change.stream = stream_gear_change
+
+	car_audio_player_wind.volume_db = -50
+	car_audio_player_wind.play()
+	
+	
+	high_pass_filter = AudioServer.get_bus_effect(1, 0) as AudioEffectHighPassFilter
+	low_pass_filter = AudioServer.get_bus_effect(1, 1) as AudioEffectLowPassFilter
+
+
+func _process(delta):
+	#calculate db of drag wind
+	var car_linear_velocity_ms = car_ref.linear_velocity.length()
+	car_audio_player_wind.volume_db = -50 + (car_linear_velocity_ms * 0.7)
+	
+	rpms.push_front(car_ref.rpm)
+	if rpms.size() > rpm_interpolation_frames:
+		rpms.remove_at(rpms.size() - 1)
+	rpm_average = 0.0
+	for rpm in rpms:
+		rpm_average += rpm
+	rpm_average = rpm_average / rpms.size()
+	#calculate engine db based on rpm ,rpm_curve and blend it dependent on throttle pos
+	var rpm_normalized = (car_ref.rpm - idle_rpm) / (rev_limit_rpm - idle_rpm)
+	car_audio_player_engine_on.volume_db = (db_engine_on - 20) + clamp((20 * db_blend_curve_on.sample_baked(car_ref.throttle_input)), 0, 20) + (db_rpm_curve.sample_baked(rpm_normalized))
+	car_audio_player_engine_off.volume_db = (db_engine_off - 20) + (20 * db_blend_curve_off.sample_baked(car_ref.throttle_input)) +(db_rpm_curve.sample_baked(rpm_normalized))
+
+	# calculate pitch from car rpm
+	var pitch_out = clamp(rpm_average * 0.001, pitch_min, pitch_max)
+	
+	if filter_off_throttle:
+		high_pass_filter.cutoff_hz = clamp(20 + (db_blend_curve_off.sample_baked(car_ref.throttle_input) * (high_pass_cutoff_freq - 20)), 20 , 500)
+		low_pass_filter.cutoff_hz = 20500 - (db_blend_curve_off.sample_baked(car_ref.throttle_input) * (20500 - low_pass_cutoff_freq))
+	elif filter_on_throttle:
+		high_pass_filter.cutoff_hz = clamp(20 + (db_blend_curve_on.sample_baked(car_ref.throttle_input) * (high_pass_cutoff_freq - 20)), 20 , 500)
+		low_pass_filter.cutoff_hz = 20500 - (db_blend_curve_on.sample_baked(car_ref.throttle_input) * (20500 - low_pass_cutoff_freq))
+	else :
+		high_pass_filter.cutoff_hz = 20
+		low_pass_filter.cutoff_hz = 20500
+	print("cutoff highpass= " + str(high_pass_filter.cutoff_hz))
+
+	#fade in and out the brake disc sample 
+	if car_ref.brake_input > 0.5 and car_linear_velocity_ms > 5:
+		timer_brakes += delta
+		if timer_brakes > 0.1:
+			if !car_audio_player_brakes.playing:
+				car_audio_player_brakes.play()
+			if car_audio_player_brakes.volume_db < db_brake_noise:
+				car_audio_player_brakes.volume_db += delta * (50.0 / brake_noise_fade_in_time)
+				car_audio_player_brakes.pitch_scale = brake_noise_pitch
+	else :
+		timer_brakes = 0.0
+		if car_audio_player_brakes.volume_db > -50:
+			car_audio_player_brakes.volume_db -= delta * 50
+
+	#play gear change sample when gear-change is detected
```

Comment:
I think the logic for gear-change sound could be made in to a method that gets called when gears get changed by making a gear_changed signal in car script and connecting to it.

Reply:
yes, thats true. i can do this.... i think there should be generally a overwork for shifthing, so old cars not imediatly change gears, for sequential gearboxes its ok but for Manual shifting should be a shift time, and a delay for shift sound...

Reply:
I agree with the gear change delay. Right now every car just changes gear immediately, but we could add a per-car delay, which could be set to quite high for older cars and 0 / really low for cars that has a sequential gearbox. If you could add it it would be great!